mavros_extras/src/plugins/obstacle_distance.cpp

```diff
@@ -101,7 +101,9 @@ class ObstacleDistancePlugin : public plugin::PluginBase {
 					}
 				}
 			}
-			obstacle.increment = ceil(req->angle_increment * RAD_TO_DEG * scale_factor);	//!< [degrees]
```

Comment:
It certainly looks like master has a bug in that it's not setting the obstacle.increment_f.

I'm not a mavros expert so take this with a grain of salt but where is "fixed.ranges.size()" coming from?  Is it the same as req->ranges.size?

I think the new increment_deg calculation is maybe not correct.  It seems to be using obstacle.distances.size() which (i think) is always 72 but I think it instead needs to use the number of slots in obstacle.distsances that have actually been filled in with values.

> const float increment_deg = req->angle_increment * RAD_TO_DEG * scale_factor;
> obstacle.increment = static_cast<uint8_t>(increment_deg + 0.5f);  //!< Round to nearest integer.
> obstacle.increment_f = increment_deg;

As a small style thing we could move the set for obstacle.increment_f two lines higher and remove the intermediate variable increment_deg but it doesn't really matter.  The similar code several lines higher does it this way.

Reply:
Ah, fixed was a result of my attempts to rotate the laserscan; I must have accidentally left them there. My apologies.

Looking at the code again, if you have more than 72 values in your scan, it will fill in all 72 slots in the obstacle distance. Unless you're talking about NaN values and such.

Reply:
Russell, So you agree that my suggestion for how obstacle.increment_f should be calculated is correct?  If yes maybe give it a try if you have time and update the PR?

I think the calculation of the obstacle.increment_f as it is currently in this PR will end up with the objects appearing in the wrong place on the receiver side.

Thanks again for finding this issue and trying to get it fixed.